src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,64 @@
+package duke.task;
+
+/**
+ * A {@code Task} represents a task with a description and a status (done or not done).
+ */
+public class Task {
+
+    /**
+     * The description of the {@code Task}.
+     */
+    public String description;
+
+    /**
+     * The status of the {@code Task}.
+     */
+    protected boolean isDone;
+
+    /**
+     * Creates a new {@code Task} with the given description.
+     *
+     * @param description The description of the {@code Task}.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the status icon of the {@code Task}.
+     *
+     * @return The status icon of the {@code Task}.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "[X]" : "[ ]"); // mark done task with X
+    }
+
+    /**
+     * Marks the {@code Task} as done.
+     */
+    public void markDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the {@code Task} as not done.
+     */
+    public void unmarkDone() {
```

Comment:
Personally I felt that unmarkDone is not a very intuitive name, perhaps just using unmark will be better?